@xen-orchestra/rest-api/src/servers/server.controller.mts

```
@@ -48,4 +55,46 @@ export class ServerController extends XoController<XoServer> {
   getServer(@Path() id: string): Promise<Unbrand<XoServer>> {
     return this.getObject(id as XoServer['id'])
   }
+
+  /**
+   * @example id "f07ab729-c0e8-721c-45ec-f11276377030"
+   */
+  @Example(taskLocation)
+  @Post('{id}/actions/connect')
+  @SuccessResponse(actionAsyncroneResp.status, actionAsyncroneResp.description, actionAsyncroneResp.produce)
+  @Response(noContentResp.status, noContentResp.description)
+  @Response(notFoundResp.status, notFoundResp.description)
+  @Response(409, 'The server is already connected')
+  connectServer(@Path() id: string, @Query() sync?: boolean): Promise<void | string> {
+    const serverId = id as XoServer['id']
+    const action = async () => {
+      await this.restApi.xoApp.connectXenServer(serverId)
+    }
+
+    return this.createAction<void>(action, {
+      statusCode: noContentResp.status,
+      sync,
+      taskProperties: { name: 'connect server', objectId: serverId },
+    })
+  }
+
+  /**
+   * @example id "f07ab729-c0e8-721c-45ec-f11276377030"
+   */
+  @Example(taskLocation)
+  @Post('{id}/actions/disconnect')
+  @SuccessResponse(actionAsyncroneResp.status, actionAsyncroneResp.description, actionAsyncroneResp.produce)
+  @Response(noContentResp.status, noContentResp.description)
+  disconnectServer(@Path() id: string, @Query() sync?: boolean): Promise<void | string> {
+    const serverId = id as XoServer['id']
+    const action = async () => {
+      await this.restApi.xoApp.disconnectXenServer(serverId)
+    }
```

connect server fails if server is already connected, but disconnectXenServer silently accept to disconnect an already disconnected server. Is it expected ?